client/state/activity-log/utils.js

```diff
@@ -3,22 +3,45 @@
 export const filterStateToApiQuery = filter =>
 	Object.assign(
 		{},
-		{ number: 1000 },
-		filter.group && filter.group.includes && { group: filter.group.includes }
+		filter.action && { action: filter.action },
+		filter.after && { after: filter.after },
+		filter.before && { before: filter.before },
+		filter.by && { by: filter.by },
+		filter.dateRange && { date_range: filter.dateRange },
+		filter.group && filter.group.includes && { group: filter.group.includes },
+		filter.group && filter.group.excludes && { not_group: filter.group.excludes },
+		filter.name && { name: filter.name },
+		{ number: 1000 }
 	);
 
 export const filterStateToQuery = filter =>
 	Object.assign(
 		{},
-		filter.page > 1 && { page: filter.page },
-		filter.group && filter.group.includes && { group: filter.group.includes.join( ',' ) }
+		filter.action && { action: filter.action.join( ',' ) },
+		filter.after && { after: filter.after },
+		filter.before && { before: filter.before },
+		filter.by && { by: filter.by },
+		filter.dateRange && { date_range: filter.dateRange },
+		filter.group && filter.group.includes && { group: filter.group.includes.join( ',' ) },
+		filter.group && filter.group.excludes && { not_group: filter.group.excludes( ',' ) },
+		filter.name && { name: filter.name.join( ',' ) },
+		filter.page > 1 && { page: filter.page }
 	);
 
 export const queryToFilterState = query =>
 	Object.assign(
 		{},
-		query.page && query.page > 0 && { page: query.page },
+		query.action && { action: decodeURI( query.action ).split( ',' ) },
+		query.after && { after: query.after },
+		query.before && { before: query.before },
+		query.by && { by: query.by },
+		query.date_range && { dateRange: query.date_range },
```

Comment:
is this still the string value that isn't really a range but more a description of a date, like `two weeks ago`?

do we want to perform any transformation here as it comes into Calypso or will it end up showing as plain text in a query bar somewhere and never really carry any more meaning from inside?

Reply:
No transforms necessary at the moment. There are no UI designs for this parameter yet, but i've added it based on hangout discussions.